code/go/0chain.net/blobbercore/handler/handler.go

```diff
@@ -153,6 +153,17 @@ func WithStatusConnection(handler common.StatusCodeResponderF) common.StatusCode
 	}
 }
 
+func WithStatusReadOnlyConnection(handler common.StatusCodeResponderF) common.StatusCodeResponderF {
+	return func(ctx context.Context, r *http.Request) (interface{}, int, error) {
+		ctx = GetMetaDataStore().CreateTransaction(ctx)
+		resp, statusCode, err := handler(ctx, r)
+		defer func() {
+			GetMetaDataStore().GetTransaction(ctx).Rollback()
```

Comment:
If the transaction was used by other threads and update the database, what would happen by calling the `Rollback()` here? Is it possible the changes will be reset?

Reply:
yes. it was by designed. it is copied from old `WithReadonlyConnection` method

Reply:
I mean if we have two API endpoints, and both call the `GetMetaDataStore().CreateTransaction()` to get the ctx, suppose endpoint A update the database, but not commit the changes yet, then another API endpoint called the rollback() in the defer function above to the same ctx, will the data changes for endpoint A be  reset?